### PR TITLE
Search Block: Master PR

### PR DESCRIPTION
**Pull request by AmandaRiu**
Related to #3073. Instead of creating a new PR for every PR reviewed in Gutenberg, Cameron and I thought it'd make more sense to just have a master PR for updating the gutenberg reference for testing but not merging until the Search block has been added to the `initial-html.js` or has breaking tests. This way this PR can act as a record of all the PRs and it will be easier to parse. 
- The last entry in the table below with the ☑️ **will be the active gutenberg reference under test**. 
- Once the gutenberg PR has been merged it will be updated with the ✅  symbol. 

**This PR is now ready for merge once the last PR in the table below has been approved and merged**

Gutenberg PR | Merged
-- | --
[Search Block - Add to inserter](https://github.com/WordPress/gutenberg/pull/28983) | ✅
[Search Block: Add block toolbar options](https://github.com/WordPress/gutenberg/pull/28985) | ✅
[Search Block: Basic UI](https://github.com/WordPress/gutenberg/pull/29045) | ✅ 
[Search Block: Add button position UI support](https://github.com/WordPress/gutenberg/pull/29096) | ✅ 
[Search Block: Width Setting](https://github.com/WordPress/gutenberg/pull/29275) | ✅
[Search Block: Fix iOS Focus Issues](https://github.com/WordPress/gutenberg/pull/29500) | ✅ 
[Search Block: Move Block Toolbar options to settings panel](https://github.com/WordPress/gutenberg/pull/29541) | ✅ 
[Search block: Styling](https://github.com/WordPress/gutenberg/pull/30052) | ✅ 
[Search block: Accessibility](https://github.com/WordPress/gutenberg/pull/30216) | ✅ 
[Search block: Unit Tests](https://github.com/WordPress/gutenberg/pull/30511) | ✅ 
[Search block: Long button text handling](https://github.com/WordPress/gutenberg/pull/30855) | ✅ 
[Search block: UI Tests](https://github.com/WordPress/gutenberg/pull/30783) | ✅ 

### To test
Click the active Gutenberg PR in the table to view full test instructions. 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 Unreleased
 ------
+* [***] New Block: Search Block [#3210]
+* a11y: Bug fix: Allow stepper cell to be selected by screenreader [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3362]
```
Comment (dcalhoun) [inline RELEASE-NOTES.txt]: I believe this line may have been erroneously added in a merge conflict resolution. It appears to already be present further down in this file for the `1.51.0` release.

Comment (AmandaRiu) [reply RELEASE-NOTES.txt]: Good catch! Fixed in 524231751c252dcd6c4069da6d73a7514b6e1737

**Comment by guarani** [comment]
Being new on the block 😏, I think the search block gets an invite to the "[other block list](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/src/block-support/supported-blocks.json)", as suggested [here](https://github.com/WordPress/gutenberg/blob/0037ab3d6cd986b730924a36720a2583e3b7933e/packages/block-library/src/index.native.js#L218).
I think it's easy to miss so thought I'd mention it in case it's not on your radar.

**Comment by dcalhoun** [comment]
@AmandaRiu I attempted to test this branch within WPiOS via setting [the `gutenberg` ref](https://github.com/wordpress-mobile/WordPress-iOS/blob/f3e5ff5f29c5437697adb7daef09cec844d2a547/Podfile#L165) to this branch's last commit. I also attempted by checking out this branch, `npm run bundle`, and `LOCAL_GUTENBERG=true bundle exec pod install`.  However, the Search block would not show up within the block inserter menu. Am I overlooking something?

**Comment by AmandaRiu** [comment]
Oh thank you @guarani  for this information:  

> Being new on the block 😏, I think the search block gets an invite to the "[other block list](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/src/block-support/supported-blocks.json)", as suggested [here](https://github.com/WordPress/gutenberg/blob/0037ab3d6cd986b730924a36720a2583e3b7933e/packages/block-library/src/index.native.js#L218).
> I think it's easy to miss so thought I'd mention it in case it's not on your radar.

I had completely missed that.

**Comment by AmandaRiu** [comment]
> @AmandaRiu I attempted to test this branch within WPiOS via setting [the `gutenberg` ref](https://github.com/wordpress-mobile/WordPress-iOS/blob/f3e5ff5f29c5437697adb7daef09cec844d2a547/Podfile#L165) to this branch's last commit. I also attempted by checking out this branch, `npm run bundle`, and `LOCAL_GUTENBERG=true bundle exec pod install`. However, the Search block would not show up within the block inserter menu. Am I overlooking something?

@dcalhoun I think it may have been because I hadn't added the search block to the supported blocks in gutenberg mobile as @guarani had pointed out in [this comment.](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3210#issuecomment-825349014). I've added it now though so it should work 🤞

**Review by dcalhoun — APPROVED** [review]
LGTM. :shipit: I tested this, with a focus on https://github.com/WordPress/gutenberg/pull/30783, on an iPhone SE and Samsung Galaxy S20. 

We'll just need to change the `gutenberg` ref to a merge commit before merging this work.

**Pull request merged**